### PR TITLE
Replace URL detection with autolink-java

### PR DESCRIPTION
**Pull request by ChrisJohnNOAA**
# Description

I've started using CodeQL to highlight code concerns (it has a focus on security and performance issues). CodeQL highlighted the url extraction regexes as potentially extremely slow due to backtracking. While looking into fixes, I came across a library that does a better job than the regexes (there's some edge cases it handles correctly) and doesn't trigger CodeQL warnings.

Since I don't believe we have legitimate performance concerns with either approach I did not run a performance test.

Fixes CodeQL warnings. These aren't enabled on the main ERDDAP repo yet, and likely show up as security issues so I don't think I could link to them.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: WEB-INF/classes/com/cohort/util/LinkHelper.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
+package com.cohort.util;
+
+import java.util.EnumSet;
+import org.nibor.autolink.LinkExtractor;
```
Comment (srstsavage) [inline WEB-INF/classes/com/cohort/util/LinkHelper.java:4]: Love the isolation of the autolink dependency to this `LinkHelper` class in case it ever needs to be replaced.

**Review by srstsavage — APPROVED** [review]
